### DISCUSSION TITLE
Временная таблица

### DISCUSSION THREAD

**Discussion by Vadim017** [totumonline/totum-ru-issues-and-discussions#588] [Как это сделать?] [answered]
У меня идёт импорт данных из простой таблицы во временную таблицу, в ней сделал специальное поле в конце (izmenenie), которое меняет своё значение при изменении строки для того чтоб понимать в каких строчках было сделано изменение и именно по этому полю система понимала какие строчки надо извлечь из временной таблицы и отправить в простую таблицу. В простой таблице есть ID строк, но во время импорта они почему то сбиваются поэтому я вместе с данными передаю ID в поле ID_nom, чтоб у строк оставалось индивидуальное поле. Что можно сделать чтоб оно работало? помогите пожалуйста 

В итоге тотум выдаёт ошибку "В параметре where необходимо использовать лист по количеству изменяемых строк либо не лист.

```
a1=: setListExtended(table: 'zayavki_na_pech_prod'; field: 'fio_zayavitelya' = $fio_zayavitelya_ch; where: 'id' = $tid_ch;  log: true)
	
		
		fio_zayavitelya_ch: listSection(list: $fio_zayavitelya_list_ch; item: "fio_zayavitelya")
		fio_zayavitelya_list_ch: selectRowList(table: $#ntn; field: 'fio_zayavitelya'; order: 'id_nom' asc; where: 'izmenenie' = "E")
	
	
		tid_ch: listSection(list: $tid_list_ch; item: "id_nom")
		tid_list_ch: selectRowList(table: $#ntn; field: 'id_nom'; order: 'id' asc)
```

[test.txt](https://github.com/user-attachments/files/16674547/test.txt)

Функция setListExtended; action поле h_izmenenie_zapisey таблицы pechatnaya_produkciya"

**Comment by ttm-alexeypolunin** [comment, marked as answer]
У вас разная размерность получается. В значение идет список фильтрованный по `E`, а в `id_nom` все строки временной таблицы.  Об этом и ошибка пишет, что размерность не совпадает «необходимо использовать лист по количеству изменяемых строк».

Я упрощу вам код:

```
a1=: setListExtended(table: 'simple_table'; field: 'some_field' = $rowlist[[new_value_field]]; where: 'id' = $rowlist[[id_nom]];  log: true)

~rowlist: selectRowList(table: $#ntn; field: 'new_value_field'; field: 'id_nom';  where: 'change' = "E"; order: 'id_nom' asc;)
```

~ тильда в таких случаях нужна, что бы не запрашивать двараза БД про одно и тоже, иначе суть выполнения  selectRowList теряется
